### ISSUE TITLE
VChat option type are broken

### ISSUE THREAD

**Issue by kaine0923** [ecomfe/vue-echarts#751]
### 请确认

- [X] 我可以确认这个问题无法在 ECharts 项目本身中复现。

### 您是如何将 Vue-ECharts 引入项目的？

通过 ES 模块 import

### 版本信息

```sh
+-- vue-echarts@6.6.2 
| +-- echarts@5.4.3 
| +-- vue-demi@0.13.11
| `-- vue@3.3.10
```

### 问题详情

![image](https://github.com/ecomfe/vue-echarts/assets/121680374/956ca66e-67a0-45dd-84b7-a62ea34993b1)

https://github.com/ecomfe/vue-echarts/blob/e224c49951d148d3b0dc300e2b5640a7ea7d1540/src/types.ts#L26

https://github.com/apache/echarts/blob/master/src/export/core.ts#L96-L113

```typescript
[key in GetMainType<OptionUnion>]?: Arrayable<ExtractComponentOption<OptionUnion, key>>
``` 
The purpose of this code is to retrieve the option type for a specific component in EChartsOption. However, due to the type inference, the code produces the following inference result
```typescript
[x: string]: unknown
[x: number]: unknown
```

### 问题复现

https://github.com/kaine0923/vue-echarts-type-issue

**Comment by liyuan1125** [comment]
+1

**Comment by radusuciu** [comment]
Can confirm this issue happens with versions > 6.6.1. Option typing and inference worked fine for me before but is now broken.

**Comment by DMOAbove** [comment]
Same here, after 6.6.1 as previously said.

![image](https://github.com/ecomfe/vue-echarts/assets/122013699/b5fd84b6-cd20-48b6-a277-18cf633586d6)

**Comment by demiro** [comment]
I have the same issue... and can confirm that downgrading it to 6.6.1 works

**Comment by Justineo** [comment]
Seems to be introduced by https://github.com/ecomfe/vue-echarts/commit/e224c49951d148d3b0dc300e2b5640a7ea7d1540.

I've reverted the changes and it should be now fixed in v6.6.5.
